### PR TITLE
Convert PlayerBots tables to InnoDB

### PR DESCRIPTION
**Pull request by elderbit**
Convert PlayerBots tables to InnoDB (disable strict mode during conversion)

# Pull Request

### This change converts the PlayerBots-related tables from MyISAM to InnoDB.

**Why this is beneficial (even without fixing a specific bug):**

- Crash safety & data integrity: InnoDB is transactional and uses redo logs; it provides automatic crash recovery, unlike MyISAM which can require manual repairs after unclean shutdowns.
- Row-level locking: InnoDB reduces write contention and improves concurrency under bot-heavy workloads compared to MyISAM’s table-level locks.
- Consistent reads: InnoDB supports MVCC, enabling stable reads while writes are happening—useful for mixed read/write access patterns.
- Operational robustness: Better behavior under backup/restore and replication scenarios; fewer “table marked as crashed” style issues.

Strict mode handling:
The migration toggles innodb_strict_mode off only for the session to prevent the conversion from failing on edge-case legacy definitions, then re-enables it immediately after.

---

## How to Test the Changes

- Step-by-step instructions to test the change
Run the SQL script in the Playerbot database.
- Any required setup (e.g. multiple players, bots, specific configuration)
No
- Expected behavior and how to verify it
All tables should now have been converted from InnoDB to MyISAM.
This script should return nothing:

```
SELECT
    t.TABLE_SCHEMA AS db_name,
    t.TABLE_NAME   AS table_name,
    t.ENGINE       AS storage_engine
FROM information_schema.TABLES t
WHERE t.TABLE_SCHEMA = DATABASE()
-- With phpMyAdmin, use the following and insert your database name, e.g., “acore_playerbots.”
-- WHERE t.TABLE_SCHEMA = 'YOUR_PLAYERBOT_DB'
  AND t.TABLE_TYPE = 'BASE TABLE'
  AND t.ENGINE = 'MyISAM'
ORDER BY t.TABLE_NAME;
```

## Complexity & Impact

- Does this change add new decision branches?
    - [x] No
    - [ ] Yes (**explain below**)

- Does this change increase per-bot or per-tick processing?
    - [x] No
    - [ ] Yes (**describe and justify impact**)

- Could this logic scale poorly under load?
    - [x] No
    - [ ] Yes (**explain why**)

---

## Defaults & Configuration

- Does this change modify default bot behavior?
    - [x] No
    - [ ] Yes (**explain why**)

---

## AI Assistance

- Was AI assistance (e.g. ChatGPT or similar tools) used while working on this change?
    - [x] No
    - [ ] Yes (**explain below**)

---

## Final Checklist

- [x] Stability is not compromised
- [x] Performance impact is understood, tested, and acceptable
- [ ] Documentation updated if needed
- [x]  I tested this script on a server with 2000 bots for 6 days (running 24/h) and had no issues with it.

---

## Notes for Reviewers

Anything that significantly improves realism at the cost of stability or performance should be carefully discussed
before merging.

### CODE REVIEW

File: data/sql/playerbots/updates/2026_01_30_00_change_to_InnoDB.sql
Comment (kadeshar) [inline data/sql/playerbots/updates/2026_01_30_00_change_to_InnoDB.sql:1]: Missing file extension

Comment (elderbit) [reply data/sql/playerbots/updates/2026_01_30_00_change_to_InnoDB.sql:1]: Oops, fixed.

File: data/sql/playerbots/updates/2026_01_30_00_change_to_InnoDB.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+-- Temporarily disables innodb_strict_mode for the session to allow the script to complete even if legacy table definitions contain InnoDB-incompatible attributes
+SET SESSION innodb_strict_mode = 0;	
+
+-- Change the tables to InnoDB
+ALTER TABLE playerbots_guild_names ENGINE=InnoDB;
+ALTER TABLE playerbots_names ENGINE=InnoDB;
```
Comment (kadeshar) [inline data/sql/playerbots/updates/2026_01_30_00_change_to_InnoDB.sql]: That is characters database not playerbots. Sorry i noticed that by accident right now

Comment (elderbit) [reply data/sql/playerbots/updates/2026_01_30_00_change_to_InnoDB.sql]: Sorry, I should have been more careful. 
Fixed.

**Comment by NoxMax** [comment]
updates and updates_include too?

```
+------------------+-------------------------------------------+----------------+
| db_name          | table_name                                | storage_engine |
+------------------+-------------------------------------------+----------------+
| acore_playerbots | playerbots_dungeon_suggestion_abbrevation | MyISAM         |
| acore_playerbots | playerbots_dungeon_suggestion_definition  | MyISAM         |
| acore_playerbots | playerbots_dungeon_suggestion_strategy    | MyISAM         |
| acore_playerbots | playerbots_equip_cache                    | MyISAM         |
| acore_playerbots | playerbots_item_info_cache                | MyISAM         |
| acore_playerbots | playerbots_rarity_cache                   | MyISAM         |
| acore_playerbots | playerbots_rnditem_cache                  | MyISAM         |
| acore_playerbots | playerbots_tele_cache                     | MyISAM         |
| acore_playerbots | playerbots_travelnode                     | MyISAM         |
| acore_playerbots | playerbots_travelnode_link                | MyISAM         |
| acore_playerbots | playerbots_travelnode_path                | MyISAM         |
| acore_playerbots | updates                                   | MyISAM         |
| acore_playerbots | updates_include                           | MyISAM         |
+------------------+-------------------------------------------+----------------+
```

**Comment by elderbit** [comment]
> updates and updates_include too?
> ...


I had problems with the two tables you mentioned, so I removed them from the PR until I had tested and understood it in detail.

**Review by hermensbas — APPROVED** [review]
